ImperatorToCk3Tests/ImperatorWorldTests/Characters/CharacterTests.cpp

```diff
@@ -0,0 +1,379 @@
+
+	ASSERT_EQ("paradoxian", theCharacter.getReligion());
+}
+
+TEST(ImperatorWorld_CharacterTests, religionDefaultsToBlank)
+{
+	std::stringstream input;
+	input << "=\n";
+	input << "{\n";
+	input << "}";
+
+	const ImperatorWorld::Character theCharacter(input, 42);
+
+	ASSERT_TRUE(theCharacter.getReligion().empty());
+}
+
+TEST(ImperatorWorld_CharacterTests, sexCanBeSet)
+{
+	std::stringstream input;
+	input << "=\n";
+	input << "{\n";
+	input << "\tfemale=yes";
+	input << "}";
+
+	const ImperatorWorld::Character theCharacter(input, 42);
+
+	ASSERT_TRUE(theCharacter.isFemale());
+}
+TEST(ImperatorWorld_CharacterTests, sexDefaultsToMale)
+{
+	std::stringstream input;
+	input << "=\n";
+	input << "{\n";
+	input << "}";
+
+	const ImperatorWorld::Character theCharacter(input, 42);
+
+	ASSERT_FALSE(theCharacter.isFemale());
+}
+
+TEST(ImperatorWorld_CharacterTests, traitsCanBeSet)
+{
+	std::vector<std::string> traitsVector{ "lustful", "submissive", "greedy" };
+	
+	std::stringstream input;
+	input << "=\n";
+	input << "{\n";
+	input << "\ttraits = { \"lustful\" \"submissive\" \"greedy\" }";
+	input << "}";
+
+	const ImperatorWorld::Character theCharacter(input, 42);
+
+	ASSERT_EQ(traitsVector, theCharacter.getTraits());
+}
+
+TEST(ImperatorWorld_CharacterTests, traitsDefaultToEmpty)
+{
+	std::stringstream input;
+	input << "=\n";
+	input << "{\n";
+	input << "}";
+
+	const ImperatorWorld::Character theCharacter(input, 42);
+
+	ASSERT_TRUE(theCharacter.getTraits().empty());
+}
+
+TEST(ImperatorWorld_CharacterTests, birthDateCanBeSet)
+{
+	std::stringstream input;
+	input << "=\n";
+	input << "{\n";
+	input << "\tbirth_date=408.6.28";
+	input << "}";
+
+	const ImperatorWorld::Character theCharacter(input, 42);
+
+	ASSERT_EQ(date("408.6.28"), theCharacter.getBirthDate());
+}
+
+TEST(ImperatorWorld_CharacterTests, birthDateDefaultsTo1_1_1)
+{
+	std::stringstream input;
+	input << "=\n";
+	input << "{\n";
+	input << "}";
+
+	const ImperatorWorld::Character theCharacter(input, 42);
+
+	ASSERT_EQ(date("1.1.1"), theCharacter.getDeathDate());
+}
+
+TEST(ImperatorWorld_CharacterTests, deathDateCanBeSet)
+{
+	std::stringstream input;
+	input << "=\n";
+	input << "{\n";
+	input << "\tdeath_date=408.6.28";
+	input << "}";
+
+	const ImperatorWorld::Character theCharacter(input, 42);
+
+	ASSERT_EQ(date("408.6.28"), theCharacter.getDeathDate());
+}
+
+TEST(ImperatorWorld_CharacterTests, deathDateDefaultsTo1_1_1)
+{
+	std::stringstream input;
+	input << "=\n";
+	input << "{\n";
+	input << "}";
+
+	const ImperatorWorld::Character theCharacter(input, 42);
+
+	ASSERT_EQ(date("1.1.1"), theCharacter.getBirthDate());
+}
+
+TEST(ImperatorWorld_CharacterTests, spousesCanBeSet)
+{
+	std::stringstream input;
+	input << "=\n";
+	input << "{\n";
+	input << "\tspouse= { 69 420 } ";
+	input << "}";
+
+	const ImperatorWorld::Character theCharacter(input, 42);
+
+	ASSERT_FALSE(theCharacter.getSpouses().empty());
+	ASSERT_EQ(theCharacter.getSpouses().begin()->first, 69);
```

Comment:
maps don't have order. It may be the first, it may not.

Reply:
Done, also changed to be consistent with the `value, function` order.